### PR TITLE
LPS-164396 [Raylife AP] Creating APPLICATIONS SEARCH

### PR DESCRIPTION
**Pull request by eduardo-diniz**
TASK: [LPS-164396](https://issues.liferay.com/browse/LPS-164396)

### CODE REVIEW

**Comment by eduardo-diniz** [comment]
[TASK: LPS-152415](https://issues.liferay.com/browse/LPS-152415)

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
Just starting reviewing :)

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Review by eliasantosf — CHANGES_REQUESTED** [review]
Some small suggestions.

**Comment by eduardo-diniz** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx [hunk 1]
```diff
@@ -96,8 +92,43 @@ const ApplicationsTable = () => {
...
+	const lastNameSearched = searchInput.split(' ')[1];
+
+	const PARAMETERS_DEFAULT = {
+		page: '0',
+		pageSize: '0',
+		sort: 'applicationCreateDate:desc',
+	};
+	const PARAMETERS_SEARCH = {
+		filter:
+			lastNameSearched === undefined
```
[the remark this answers is not in the document]
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx]: thx

**Review by eliasantosf — APPROVED** [review]
LGTM :fire: Thanks @eduardo-diniz

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by pedroliveirav** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx [hunk 2]
```diff
@@ -96,8 +92,41 @@ const ApplicationsTable = () => {
...
+	const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+		setSearchInput(event.target.value);
+	};
+	const handleClick = () => {
+		if (!searchInput) {
+			setParameters(PARAMETERS_DEFAULT);
+		}
+
+		return setParameters(PARAMETERS_SEARCH);
+	};
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx:129]: Please, change it.
```suggestion
	const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
		setSearchInput(event.target.value);
	};
	
	const handleClick = () => {
		if (!searchInput) {
			setParameters(PARAMETERS_DEFAULT);
		}

		return setParameters(PARAMETERS_SEARCH);
	};
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx [hunk 3]
```diff
@@ -96,8 +92,41 @@ const ApplicationsTable = () => {
...
+		sort: 'applicationCreateDate:desc',
+	};
+	const PARAMETERS_SEARCH = {
+		filter: !lastNameSearched
+			? `contains(firstName,'${firstNameSearched}') or contains(lastName,'${firstNameSearched}') or contains(email, '${firstNameSearched}') or contains(externalReferenceCode, '${firstNameSearched}')`
+			: `contains(firstName,'${firstNameSearched}') and contains(lastName,'${lastNameSearched}')`,
+		page: '0',
+		pageSize: '0',
+		sort: 'applicationCreateDate:desc',
+	};
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx:111]: Please, change it.

```suggestion
	const PARAMETERS_DEFAULT = {
		page: '0',
		pageSize: '0',
		sort: 'applicationCreateDate:desc',
	};
	
	const PARAMETERS_SEARCH = {
		filter: !lastNameSearched
			? `contains(firstName,'${firstNameSearched}') or contains(lastName,'${firstNameSearched}') or contains(email, '${firstNameSearched}') or contains(externalReferenceCode, '${firstNameSearched}')`
			: `contains(firstName,'${firstNameSearched}') and contains(lastName,'${lastNameSearched}')`,
		page: '0',
		pageSize: '0',
		sort: 'applicationCreateDate:desc',
	};
```

File: modules/apps/site-initializer/site-initializer-raylife-d2c/src/main/resources/site-initializer/object-entries/1_application.json [hunk 4]
```diff
@@ -55,16 +55,16 @@
 			"productName": "Home"
 		},
 		{
-			"applicationCreateDate": "2022-09-09",
+			"applicationCreateDate": "2022-09-09T13:00:00Z",
 			"applicationStatus": "quoted",
 			"email": "joana@liferay.com",
 			"externalReferenceCode": "RAYAPP-007",
-			"firstName": "Joana",
+			"firstName": "Eduardo",
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-d2c/src/main/resources/site-initializer/object-entries/1_application.json:62]: Why do you change it?
```suggestion
			"firstName": "Joana",
```

**Review by silasbsi — CHANGES_REQUESTED** [review]
I added some suggestions.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx (hunk 2 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx:129]: done

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx (hunk 3 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/pages/ApplicationsTable/index.tsx:111]: done, thx!

File: modules/apps/site-initializer/site-initializer-raylife-d2c/src/main/resources/site-initializer/object-entries/1_application.json (hunk 4 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-d2c/src/main/resources/site-initializer/object-entries/1_application.json:62]: Just for testing the sort date functionality.

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Review by silasbsi — APPROVED** [review]

**Comment by silasbsi** [comment]
ci:forward

**Comment by silasbsi** [comment]
ci:forward

**Comment by eliasantosf** [comment]
ci:forward

**Pull request closed without merging**